pinot-spi/src/main/java/org/apache/pinot/spi/eventlistener/query/PinotBrokerQueryEventListenerFactory.java

```diff
@@ -44,6 +50,8 @@ private PinotBrokerQueryEventListenerFactory() {
   public synchronized static void init(PinotConfiguration eventListenerConfiguration) {
     // Initializes BrokerQueryEventListener.
     initializeBrokerQueryEventListener(eventListenerConfiguration);
+    // Initializes request headers
+    initializeAllowlistQueryRequestHeaders(eventListenerConfiguration);
```

Comment:
it is likely wont work. the prefix for this factory init is `EVENT_LISTENER_CONFIG_PREFIX="pinot.broker.event.listener"`

however the request header is associated with RequestContext which is created in PinotClientRequest. --> seems like they shouldn't share the same configuration prefix? 

if so we probably should use `pinot.broker` as prefix instead of under event listenser?

Reply:
> however the request header is associated with RequestContext which is created in PinotClientRequest. --> seems like they shouldn't share the same configuration prefix? if so we probably should use pinot.broker as prefix instead of under event listenser?

hmm I am setting these tracked-headers in `RequestHandler` classes rather than in `PinotClientRequest`. If we want to change the scope to just `pinot.broker` then does it make sense to init this conf during `BrokerStarter` rather than event-listener init. That way, we will have to pass it as an additional param in `RequestHandler` classes too.

Reply:
yeah. i thought about that e.g. the RequestContext is extended by RequestScope --> which is a spi.trace class. which might be weird to use a config key prefix to event listener. let's think it through. 

BTW the impl is fine. just need to list out exactly how they are being configured and which overrides what